index.js

```diff
@@ -22,6 +22,11 @@ function WaitForAll(count, allDone) {
 
 module.exports = function (PIXI)
 {
+    var PIXI_VERSION = Number(PIXI.VERSION.charAt(0));
+    if(PIXI_VERSION < 4) {
+        throw new Error("PIXI version needs to be >= v4.0.0");
+    }
+
     return function (resource, next)
```

Comment:
This feels like it could be broken down into smaller functions/steps.

Mainly to avoid a lot of the setting variables then overriding/compounding them with later code paths